src/Temporalio.Extensions.Hosting/TemporalWorkerServiceOptionsBuilder.cs

```diff
@@ -32,13 +33,30 @@ public TemporalWorkerServiceOptionsBuilder(string taskQueue, string? buildId, IS
             Services = services;
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TemporalWorkerServiceOptionsBuilder" />
+        /// class.
+        /// </summary>
+        /// <param name="taskQueue">Task queue for the worker.</param>
+        /// <param name="deploymentOptions">Deployment options for the worker.</param>
+        /// <param name="services">Service collection being configured.</param>
+        public TemporalWorkerServiceOptionsBuilder(string taskQueue, WorkerDeploymentOptions? deploymentOptions, IServiceCollection services)
+        {
+            TaskQueue = taskQueue;
+            DeploymentOptions = deploymentOptions;
+            Services = services;
+        }
+
         /// <inheritdoc />
         public string TaskQueue { get; private init; }
 
         /// <inheritdoc />
         public string? BuildId { get; private init; }
```

Comment:
I think we should deprecate all uses of `BuildId` in the DI stuff if it's deprecated for the non-DI stuff